homeassistant/components/cloud/onboarding.py

```diff
@@ -0,0 +1,115 @@
+"""Cloud onboarding views."""
+
+from __future__ import annotations
+
+from collections.abc import Callable, Coroutine
+from functools import wraps
+from http import HTTPStatus
+from typing import TYPE_CHECKING, Any, Concatenate
+
+from aiohttp import web
+from aiohttp.web_exceptions import HTTPUnauthorized
+
+from homeassistant.components.http import KEY_HASS
+from homeassistant.components.onboarding import (
+    _BaseOnboardingView,
+    _NoAuthBaseOnboardingView,
+)
+from homeassistant.core import HomeAssistant
+
+from . import http_api as cloud_http
+from .const import DATA_CLOUD
+
+if TYPE_CHECKING:
+    from homeassistant.components.onboarding import OnboardingStoreData
+
+
+async def async_setup_views(hass: HomeAssistant, data: OnboardingStoreData) -> None:
+    """Set up the cloud views."""
+
+    def with_cloud[_ViewT: _BaseOnboardingView, **_P](
+        func: Callable[
+            Concatenate[_ViewT, web.Request, _P],
+            Coroutine[Any, Any, web.Response],
+        ],
+    ) -> Callable[
+        Concatenate[_ViewT, web.Request, _P], Coroutine[Any, Any, web.Response]
+    ]:
+        """Home Assistant API decorator to check onboarding and cloud."""
+
+        @wraps(func)
+        async def _with_cloud(
+            self: _ViewT,
+            request: web.Request,
+            *args: _P.args,
+            **kwargs: _P.kwargs,
+        ) -> web.Response:
+            """Check onboarding status, cloud and call function."""
+            if self._data["done"]:
+                # If at least one onboarding step is done, we don't allow accessing
+                # the cloud onboarding views.
+                raise HTTPUnauthorized
+
+            hass = request.app[KEY_HASS]
+            if DATA_CLOUD not in hass.data:
+                return self.json(
+                    {"code": "cloud_disabled"},
+                    status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+                )
+
+            return await func(self, request, *args, **kwargs)
+
+        return _with_cloud
+
+    class CloudForgotPasswordView(
```

Comment:
Can we define the views on the module level?

Reply:
Yes, sure. I just kept it like this for a clean diff with the code moved from `onboarding/views.py`